src/components/Form/Form.js

```diff
@@ -0,0 +1,129 @@
+import { Link, useLocation } from 'react-router-dom';
+import { useState } from 'react';
+const Form = () => {
+
+    const location = useLocation().pathname
+    const [userName, setUserName] = useState('')
+    const [first, setFirst] = useState('')
+    const [last, setLast] = useState('')
+    const [address, setAddress] = useState('')
+    const [city, setCity] = useState('')
+    const [zip, setZip] = useState('')
+   
+    const handleSubmit = e => {
+        if (location === '/newuser' && first && last && address && city && zip.length === 5) {
+            clearInputs()
+        } else if (location === '/login' && userName) {
+            clearInputs()
+        } else if (location === '/newlocation' && address && city && zip.length === 5) {
+            clearInputs()
+        } else {
+            e.preventDefault()
+        }
```

Comment:
won't we need e.preventDefault() for all of these conditions? should we stick it at the top?

Reply:
Having `e.preventDefault()` at the top was preventing it from routing correctly, so I think the `Link` replaced the default action of the button to just change the route. We may have to change it in the future, but it is working as is currently.

Reply:
oh ok, that's right, I remember that routing prevented refreshing and makes for a smoother UX. That makes sense thanks